tests/Unit/FileTest.php

```diff
@@ -0,0 +1,236 @@
+
+        // Requesting a stream for a mime_type that has not been set
+        // should return null.
+        $this->assertNull($newFile->getStream('image/png'));
+    }
+
+    /** @test */
+    public function testBase64Data()
+    {
+        $file = new File();
+
+        $this->assertNull($file->getBase64Data());
+
+        $this->assertEquals($this->base64data,
+            $file
+                ->setBase64Data($this->base64data, 'application/pdf')
+                ->getBase64Data('application/pdf')
+        );
+
+        // Using getBase64Data() without giving a mime_type loops over the
+        // file's formats and returns the first base64Data it encounters.
+        $this->assertEquals($this->base64data,
+            $file
+                ->addFormat($this->pdfFormat['mime_type'], $this->pdfFormat['extension'])
+                ->getBase64Data()
+        );
+
+        // If no base64Data is set under the given mime_type,
+        // getBase64Data will look for data in other sources
+        // (streams or temporary files).
+        $newFile = new File();
+
+        $this->assertEquals($this->base64data,
+            $newFile
+                ->setStream($this->stream, 'image/jpeg')
+                ->getBase64Data('image/jpeg')
+        );
+        $this->assertEquals($this->base64data,
+            $newFile
+                ->setTemporaryFilePath($this->path, 'application/pdf')
+                ->getBase64Data('application/pdf')
+        );
+
+        // Requesting base64Data for a mime_type that has not been set
+        // should return null.
+        $this->assertNull($newFile->getBase64Data('image/png'));
+    }
+
+    /** @test */
+    public function testTemporaryFilePath()
+    {
+        $file = new File();
+
+        $this->assertNull($file->getTemporaryFilePath());
+
+        $this->assertEquals($this->path,
+            $file
+                ->setTemporaryFilePath($this->path, 'application/pdf')
+                ->getTemporaryFilePath('application/pdf')
+        );
+
+        // Using $this->getTemporaryFilePath() without giving a mime_type loops over the
+        // file's formats and returns the first path it encounters.
+        $this->assertEquals($this->path,
+            $file
+                ->addFormat($this->pdfFormat['mime_type'], $this->pdfFormat['extension'])
+                ->getTemporaryFilePath()
+        );
+
+        // If no path is set under the given mime_type,
+        // $this->getTemporaryFilePath() will look for data in other sources
+        // (streams or base64Data).
+        $streamFile = new File();
+
+        $streamPath = $streamFile
+            ->addFormat($this->pdfFormat['mime_type'], $this->pdfFormat['extension'])
+            ->setStream($this->stream, 'application/pdf')
+            ->getTemporaryFilePath('application/pdf');
+        $this->assertEquals(file_get_contents($this->path), file_get_contents($streamPath));
```

Comment:
It's weird that you're checking the contents of a path that seems to have nothing to do with what you've put into the file. It would be better to check on the actual value you've put into the stream.